kinesis.go

```diff
@@ -119,14 +120,18 @@ type jsonErrors struct {
 }
 
 func buildError(r *http.Response) error {
+	// Reading the body into a []byte because we might need to put it into an error
+	// message after having the JSON decoding fail to produce a message.
+	body, _ := ioutil.ReadAll(r.Body)
```

Comment:
Hello. I think we should not ignore `ioutil.ReadAll` error and also handle it. What do you think?